intel-retail/performance-tools
# feat: Convert benchmark.sh to python and add unit tests

## PR Checklist



## What are you changing?


## Issue this PR will close

close: #**issue_number**

## Anything the reviewer should know when reviewing this PR?

## Test Instructions if applicable

Prerequisites:
- python environment with Python 3.12.2 (can use [miniconda](https://rsd-intel.slack.com/archives/C02E1TXQS8Z/p1711055235807679?thread_ts=1711050322.426669&cid=C02E1TXQS8Z) and create a [Python 3.12.2 env](https://conda.io/projects/conda/en/latest/user-guide/tasks/manage-python.html))
- installed necessary packages listed in requirements.txt

1. Pull the latest changes from this branch and from the retail-use-cases for updated compose files
2. change directories to the `benchmark-script` directory
3.  change directories `cd docker` and run `make build-igt` to build the benchmark docker container
4. change directories back `cd ..`
5. Ensure that the benchmarking script works run something like this
    ```
    python benchmark.py --compose_file ../../retail-use-cases/use-cases/gst_capi/add_camera-simulator.yml --compose_file ../../retail-use-cases/use-cases/gst_capi/add_gst_capi_yolov5_ensemble.yml
    ```
    **Note**: it is possible to provide other input variables - use `python benchmark.py --help` to see all available args
6.  run the unit tests in benchmark_test.py
    ```
    python -m unittest -v benchmark_test.py
    ```

## If the there are associated PRs in other repositories, please link them here (i.e. intel-retail/performance-tools )
